### PR TITLE
Fix visualizations on the Kubernetes overview dashboard

### PR DESCRIPTION
**Pull request by tetianakravchenko**
Signed-off-by: Tetiana Kravchenko <tetiana.kravchenko@elastic.co>

## What does this PR do?

There are two visualizations on the `[Metrics Kubernetes] Overview dashboard` with 2 broken visualizations (`Desired Pods` and `Unavailable Pods`), reason for that: index pattern is overridden (they have "yes" checked),  but `series_Index_pattern`  is not specified:
Screenshot 2021-11-11 at 09 57 40
(note that in the picture as an index_pattern was picked `logs-*`)

index pattern shouldn't be overridden.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## Author's Checklist

- [ ]

## Related issues
- Closes https://github.com/elastic/beats/issues/28792
## Screenshots

Add here screenshots presenting:
- Kibana UI forms presenting configuration options exposed by the integration:
local setup:
Screenshot 2021-11-11 at 14 46 39

### CODE REVIEW

**Comment by MichaelKatsoulis** [comment]
@tetianakravchenko I noticed there is another visualisation with `override_index_pattern: 1`. It is the [Available pods](https://github.com/elastic/integrations/blob/master/packages/kubernetes/kibana/visualization/kubernetes-da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3.json#L66). Why not changing that one too?

**Comment by tetianakravchenko** [comment]
@MichaelKatsoulis for consistency I can adjust this visualisation as well. the reason why it works: 
index pattern is actually defined: 
Screenshot 2021-11-11 at 10 02 08

[code](https://github.com/elastic/integrations/blob/master/packages/kubernetes/kibana/visualization/kubernetes-da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3.json#L69)

**Comment by MichaelKatsoulis** [comment]
> @MichaelKatsoulis for consistency I can adjust this visualisation as well. the reason why it works: index pattern is 

Ok. Thanks for the explanation. I would add it for consistency as you said.

File: packages/kubernetes/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.4.1"
+  changes:
+    - description: Fix visualizations on the Kubernetes overview dashboard
```
Comment (MichaelKatsoulis) [inline packages/kubernetes/changelog.yml:4]: The description does not reflect what was fixed

**Review by MichaelKatsoulis — APPROVED** [review]

**Review by ChrsMark — APPROVED** [review]
👍🏼

File: packages/kubernetes/changelog.yml (hunk 1 above)
Comment (tetianakravchenko) [reply packages/kubernetes/changelog.yml]: https://github.com/elastic/integrations/pull/2151/commits/2978a9f83edcd50c5dad74c47fbefe0220ae5cea

File: packages/kubernetes/kibana/visualization/kubernetes-da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3.json [hunk 2]
```diff
@@ -63,12 +63,9 @@
                                 "type": "sum"
                             }
                         ],
-                        "override_index_pattern": 1,
                         "point_size": 1,
                         "seperate_axis": 0,
-                        "series_index_pattern": "*",
                         "series_interval": "10s",
-                        "series_time_field": "@timestamp",
```
Comment (tetianakravchenko) [inline packages/kubernetes/kibana/visualization/kubernetes-da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3.json:71]: @MichaelKatsoulis @ChrsMark fyi: I've also removed this field to align all 3 visualisations: `Desired pods`, `Available Pods` and `Unavailable Pods`.
as I understand it is a default value anyway:
Screenshot 2021-11-11 at 14 53 19

Comment (ChrsMark) [reply packages/kubernetes/kibana/visualization/kubernetes-da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3.json:71]: Hmm, I'm not sure about what this field is doing. Could you provide more context?

Comment (tetianakravchenko) [reply packages/kubernetes/kibana/visualization/kubernetes-da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3.json:71]: @ChrsMark not sure where is the best place to verify: from https://github.com/elastic/kibana/blob/main/src/plugins/vis_types/timeseries/server/lib/vis_data/get_interval_and_timefield.ts#L28: `timeField` will be set to `series.series_time_field` if `override_index_pattern` is set to true, in other case - will be set to `panel.time_field`, which is set for the visualisation - https://github.com/elastic/integrations/blob/2978a9f83edcd50c5dad74c47fbefe0220ae5cea/packages/kubernetes/kibana/visualization/kubernetes-da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3.json#L75

Comment (ChrsMark) [reply packages/kubernetes/kibana/visualization/kubernetes-da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3.json:71]: Ok, but why we remove it here? Is it related to to the issue that this PR solves or it is just an extra improvement? I'm trying to understand what will  actually be different with this change and why it was like this in the past.

Comment (tetianakravchenko) [reply packages/kubernetes/kibana/visualization/kubernetes-da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3.json:71]: no, it is not related to the issue. It was removed to make those 3 visualisations configs consistent:
Screenshot 2021-11-15 at 09 57 40
see this [comment](https://github.com/elastic/integrations/pull/2151#issuecomment-966074625) - so now `Desired pods`, `Available pods` and `Unavailable pods` have the same settings: no data view override, no override of index, no override of time field

Comment (tetianakravchenko) [reply packages/kubernetes/kibana/visualization/kubernetes-da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3.json:71]: @ChrsMark I am not sure why it was added here initially, see initial PR https://github.com/elastic/integrations/pull/70
`"Unavailable pods [Metrics Kubernetes] ECS"`: has override [enabled](https://github.com/elastic/integrations/pull/70/files#diff-5541da8de32db44f74aad1487c1245b1b420dbb4457a5052c3615e5ad1d22723R65), neither `series_time_field` nor `series_index_pattern` are set
`"Available pods [Metrics Kubernetes] ECS"`: has override [enabled](https://github.com/elastic/integrations/pull/70/files#diff-16f2dfe2c9fe54fd9b828412c6ed6c2bbaa2d6009a167ce5f0fb1cd293885eb5R65), both `"series_index_pattern"` and `"series_time_field"` are set
`"Desired pods [Metrics Kubernetes] ECS"`: has override [enabled](https://github.com/elastic/integrations/pull/70/files#diff-5b4ece45ddb4451885aaaaf0b162ee5fd3a77c87f4a7a34f994e0a87e1dba960R65). `"series_time_field": "@timestamp"` is set, no `series_index_pattern`
maybe you remember why it was set this way?

Comment (ChrsMark) [reply packages/kubernetes/kibana/visualization/kubernetes-da1ff7c0-30ed-11e7-b9e5-2b5b07213ab3.json:71]: Gotcha, thanks for elaborating on this! About history, the dashboards have been imported from Beats so this one should be there for quite a long.

**Review by ChrsMark — APPROVED** [review]

**Pull request merged**